istioctl/cmd/tag.go

```diff
@@ -131,14 +135,60 @@ injection labels.`,
 				return fmt.Errorf("failed to create Kubernetes client: %v", err)
 			}
 
-			return setTag(context.Background(), client, args[0], revision, cmd.OutOrStdout())
+			return setTag(context.Background(), client, args[0], revision, false, cmd.OutOrStdout())
 		},
 	}
 
 	cmd.PersistentFlags().BoolVar(&overwrite, "overwrite", false, overrideHelpStr)
 	cmd.PersistentFlags().StringVarP(&manifestsPath, "manifests", "d", "", mesh.ManifestsFlagHelpStr)
 	cmd.PersistentFlags().BoolVarP(&skipConfirmation, "skip-confirmation", "y", false, skipConfirmationFlagHelpStr)
 	cmd.PersistentFlags().StringVarP(&revision, "revision", "r", "", revisionHelpStr)
+	cmd.PersistentFlags().StringVarP(&webhookName, "webhook-name", "", "", webhookNameHelpStr)
+	_ = cmd.MarkPersistentFlagRequired("revision")
+
+	return cmd
+}
+
+func tagGenerateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "generate <revision-tag>",
+		Short: "Generate configuration for a revision tag to stdout",
+		Long: `Create a revision tag and output to the command's stdout. Tag an Istio control plane revision for use with namespace istio.io/rev
+injection labels.`,
+		Example: ` # Create a revision tag from the "1-8-0" revision
+	istioctl x tag generate prod --revision 1-8-0 > tag.yaml
+
+	# Apply the tag to cluster
+	kubectl apply -f tag.yaml
+
+	# Point namespace "test-ns" at the revision pointed to by the "prod" revision tag
+	kubectl label ns test-ns istio.io/rev=prod
+
+	# Rollout namespace "test-ns" to update workloads to the "1-8-0" revision
+	kubectl rollout restart deployments -n test-ns
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
```

Comment:
`args` checking can be moved to `Args:`

Reply:
Moved all arg validation to `Args`, thanks